api/src/main/java/org/pmiops/workbench/reporting/ReportingUploadService.java

```diff
@@ -9,4 +11,6 @@
  */
 public interface ReportingUploadService {
   void uploadSnapshot(ReportingSnapshot reportingSnapshot);
```

Comment:
You could go ahead and deprecate this method and the `ReportingSnapshot` class (though it's generated and that might not work). Any new payloads should use the stream approach.